activejob/lib/active_job/queue_adapter.rb

```diff
@@ -17,8 +17,6 @@ module ClassMethods
       def queue_adapter=(name_or_adapter)
         @@queue_adapter = \
           case name_or_adapter
-          when :test
```

Comment:
@rafaelfranca the only advantage is that it removes this special handling, which is a weird snowflake.

I believe this doesn't really get us further away from parallel tests - if https://github.com/rails/rails/pull/16992 is merged, each test will be able to create its own anonymous subclass, which might actually be pretty simple to do. What do you think?